crates/cli-support/src/interpret.rs

```diff
@@ -0,0 +1,155 @@
+    sp: i32,
+    mem: Vec<i32>,
+    stack: Vec<i32>,
+    descriptor: Vec<u32>,
+}
+
+impl Interpreter {
+    pub fn new(module: &Module) -> Interpreter {
+        let mut ret = Interpreter::default();
+        ret.mem = vec![0; 0x100];
+        ret.sp = ret.mem.len() as i32;
+
+        for (i, s) in module.sections().iter().enumerate() {
+            match s {
+                Section::Code(_) => ret.code_idx = Some(i),
+                _ => {}
+            }
+        }
+
+        if let Some(i) = module.import_section() {
+            ret.imports = i.functions();
+            for (i, entry) in i.entries().iter().enumerate() {
+                if entry.module() != "__wbindgen_placeholder__" {
+                    continue
+                }
+                if entry.field() != "__wbindgen_describe" {
+                    continue
+                }
+                ret.describe_idx = Some(i as u32);
+            }
+        }
+
+        if let Some(e) = module.export_section() {
+            for e in e.entries() {
+                let i = match e.internal() {
+                    Internal::Function(i) => i,
+                    _ => continue,
+                };
+                ret.name_map.insert(e.field().to_string(), *i);
+            }
+        }
+
+        return ret
+    }
+
+    pub fn interpret(&mut self, func: &str, module: &Module) -> Option<&[u32]> {
+        self.descriptor.truncate(0);
+        let idx = match self.name_map.get(func) {
+            Some(i) => *i,
+            None => panic!("{} not found in exported module", func),
+        };
+        let code = match &module.sections()[self.code_idx.unwrap()] {
+            Section::Code(s) => s,
+            _ => panic!(),
+        };
+        assert_eq!(self.sp, self.mem.len() as i32);
+        assert_eq!(self.stack.len(), 0);
+        self.call(idx, code);
+        assert_eq!(self.stack.len(), 0);
+        assert_eq!(self.sp, self.mem.len() as i32);
+        Some(&self.descriptor)
+    }
+
+    fn call(&mut self, idx: u32, code: &CodeSection) {
+        use parity_wasm::elements::Instruction::*;
+
+        let body = &code.bodies()[(idx as usize) - self.imports];
+        assert!(body.locals().len() <= 1, "too many local types");
+        let locals = body.locals()
+            .get(0)
+            .map(|i| {
+                assert_eq!(i.value_type(), ValueType::I32);
+                i.count()
+            })
+            .unwrap_or(0);
+        let mut locals = vec![0; locals as usize];
+        let before = self.stack.len();
+        for instr in body.code().elements() {
+            match instr {
+                I32Const(x) => self.stack.push(*x),
+                SetLocal(i) => locals[*i as usize] = self.stack.pop().unwrap(),
+                GetLocal(i) => self.stack.push(locals[*i as usize]),
+                Call(idx) => {
+                    if Some(*idx) == self.describe_idx {
+                        self.descriptor.push(self.stack.pop().unwrap() as u32);
```

Comment:
This is jsut inlining the imported describe function, right? A comment would be nice here, it isn't totally clear why this is special cased.